core/src/main/scala/org/apache/spark/rdd/RDD.scala

```diff
@@ -1525,7 +1526,8 @@ abstract class RDD[T: ClassTag](
     val r = this.mapPartitions { iter =>
       val text = new Text()
       iter.map { x =>
-        text.set(x.toString)
+        val value = if (x != null) x.toString else "Null"
```

Comment:
This two methods are very much the same. 

I have quickly checked what is at the end of the call chain:
org.apache.spark.rdd.PairRDDFunctions#saveAsHadoopFile

There the default value of the codec is None (as there it is wrapped within an Option). 
Would not it make sense to extract these common parts? 

If that is too much still up to the calling of `.saveAsHadoopFile[TextOutputFormat[NullWritable, Text]](path)` they are common. What is your opinion?

Reply:
Regarding the NPE I think it is quite suspicious that such a trivial thing is there. Could it be it is a feature, I mean with a single mapping null-s can be changed to whatever the user wants to see and not to Null (which could cause problems latter at the processing of the possible huge result file)? So it is like a fail-fast.

Reply:
@attilapiros I think it's good to extract the common parts of these two call, and also maybe we can just add an require as @HyukjinKwon suggested, it should be uncommon for users to map null to `Null` string, `saveAsTextFile` can treat null as an exceptional case.